app/resources/api/v2/tube_rack_resource.rb

```diff
@@ -35,11 +35,20 @@ class TubeRackResource < BaseResource
                )
       filter :purpose_id, apply: ->(records, value, _options) { records.where(plate_purpose_id: value) }
 
-      # Custom methods
-      # These shouldn't be used for business logic, and a more about
-      # I/O and isolating implementation details.
-
       # Class method overrides
+      def fetchable_fields
+        super - [:tube_locations]
+      end
+
+      # Tube locations should be received as:
+      # { A1: { uuid: 'a1_tube_uuid' }, B1: { uuid: 'b1_tube_uuid' }, ... }
+      def tube_locations=(tube_locations)
+        tube_locations.each do |coordinate, tube|
+          existing_tube = Tube.with_uuid(tube[:uuid]).last
```

Comment:
This will perform a separate database lookup for each tube, interleaved with create actions. This isn't great for performance. What I usually do in this case if lookup all tubes by uuid in a single query, and then use `.index_by(&:uuid)` to build a hash that I can look-up from in the second half.

We could also consider using RackedTube.import to bulk create the associations, especially as we don't have any callbacks/validations in place (Probably add a warning to the top of RackedTube if you go that route though).

Bulk import is provided by https://github.com/zdennis/activerecord-import not native-rails as it predates our use of Rails 6. It also has slightly nice handling of things like timestamps, although it looks like that may have been improved in rails edge.